recipes/logr/0.2.0/conanfile.py

```diff
@@ -0,0 +1,105 @@
+        elif self.options.backend == "log4cplus":
+            self.requires("log4cplus/2.0.5")
+        elif self.options.backend == "log4cplus-unicode":
+            self.requires("log4cplus/2.0.5")
+
+    def configure(self):
+        minimal_cpp_standard = "17"
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+        minimal_version = {
+            "gcc": "7",
+            "clang": "7",
+            "apple-clang": "10",
+            "Visual Studio": "16"
+        }
+        compiler = str(self.settings.compiler)
+        if compiler not in minimal_version:
+            self.output.warn(
+                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
+            self.output.warn(
+                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+            return
+
+        version = tools.Version(self.settings.compiler.version)
+        if version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["LOGR_WITH_SPDLOG_BACKEND"] = self.options.backend == "spdlog"
+        self._cmake.definitions["LOGR_WITH_GLOG_BACKEND"] = self.options.backend == "glog"
+        self._cmake.definitions["LOGR_WITH_LOG4CPLUS_BACKEND"] = self.options.backend in ["log4cplus", "log4cplus-unicode"]
+
+        self._cmake.definitions["LOGR_INSTALL"] = True
+        self._cmake.definitions["LOGR_CONAN_PACKAGING"] = True
+        self._cmake.definitions["LOGR_BUILD_TESTS"] = False
+        self._cmake.definitions["LOGR_BUILD_EXAMPLES"] = False
+        self._cmake.definitions["LOGR_BUILD_BENCHMARKS"] = False
+
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        if self.options.backend == "log4cplus" and self.options["log4cplus"].unicode:
+            raise ConanInvalidConfiguration("backend='log4cplus' requires log4cplus:unicode=False")
+        elif self.options.backend == "log4cplus-unicode" and not self.options["log4cplus"].unicode:
+            raise ConanInvalidConfiguration("backend='log4cplus-unicode' requires log4cplus:unicode=True")
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib"))
+
+    def package_id(self):
+        self.info.header_only()
```

Comment:
I don't think `header_only` is incorrect as this library will generate another package/headers if it has other options.
See https://github.com/conan-io/conan/blob/4db7337f75b9c8ba02a5ef05f5fb560a417bbfe5/conans/model/info.py#L573-L576

I think this should do the following instead:
```
    self.info.settings.clear()
```
- The options should remain as it contains the requirement.
- The requires should remain too as it contains the actual requirement (include the unicode option of log4cplus)

Reply:
This can be fixed in both version in a follow up PR